Threema Integration Docs (#44160)
home-assistant/home-assistant.github.io: source/_integrations/threema.markdown

---
title: Threema
description: Instructions on how to send Threema messages from Home Assistant.
ha_category:
  - Notifications
ha_release: 2026.4
ha_config_flow: true
ha_iot_class: Cloud Push
ha_domain: threema
ha_platforms:
  - notify

> ha_integration_type: service

Reviewer: ```suggestion
ha_integration_type: service
ha_quality_scale: bronze
```

2. Configure the integration
   - Once your Gateway ID is approved, use the Gateway ID, API secret, and the previously generated keys to complete the integration setup in Home Assistant.

During setup, you can choose between two options:

- **Add existing Gateway ID**: Enter your Gateway ID, API secret, and your private and public keys to configure an existing Threema Gateway.
- **Generate new encryption keys**: The integration generates an encryption key pair and displays them. Save the keys — they cannot be recovered later. Then register a new Gateway ID at [gateway.threema.ch](https://gateway.threema.ch) using the generated public key, and complete setup with your Gateway credentials.

{% include integrations/config_flow.md %}
{% configuration_basic %}
Gateway ID:
  description: "Your Threema Gateway ID (starts with `*` and is 8 characters total)."
API secret:
  description: "The API secret from your Threema Gateway dashboard."
Private key:
  description: "Optional for basic mode, required for end-to-end encryption mode."
Public key:
  description: "Optional for basic mode, required for end-to-end encryption mode."
{% endconfiguration_basic %}

> ## Recipients

Reviewer: ```suggestion

## Recipients
```

---
title: Threema
description: Instructions on how to send Threema messages from Home Assistant.
ha_category:
  - Notifications

> ha_release: "2026.4"

Reviewer: ```suggestion
ha_release: 2026.4
```

ha_config_flow: true
ha_iot_class: Cloud Push
ha_domain: threema
ha_platforms:

>   - image

Reviewer: ```suggestion
- notify
```
is this still accurate? I don't see an image entity added in the core PR. But there is notify.

ha_integration_type: service
ha_codeowners:
  - '@LukasQ'
---

The **Threema** {% term integration %} allows you to send end-to-end encrypted text messages from Home Assistant to [Threema](https://threema.ch) users via the [Threema Gateway](https://gateway.threema.ch) service. This integration is designed for the **E2E (end-to-end) encryption mode** of Threema Gateway, where messages are encrypted locally before being sent. If no private key is configured, the integration falls back to **basic mode** (server-side encryption).

## Prerequisites

> - A [Threema Gateway](https://gateway.threema.ch) account. For testing purposes, you can [request developer credits](https://gateway.threema.ch) from Threema — make sure to select the **E2E Gateway** configuration when requesting.

Reviewer: ```suggestion
- A [Threema Gateway](https://gateway.threema.ch) account. For testing purposes, you can [request developer credits](https://gateway.threema.ch) from Threema. Make sure to select the **E2E Gateway** configuration when making the request.
```

> - A Gateway ID (starts with `*`) — either an existing one or a new one created during setup.

Reviewer: ```suggestion
- A Gateway ID (starts with `*`). You can use an existing ID or create a new one during setup.
```

- An API secret from the Threema Gateway dashboard.
- Message credits on your Threema Gateway account. Sending a message costs **1,600 credits with E2E mode** or **800 credits with basic mode**.

> **Note:** Setting up Threema Gateway is a two-step process:

Reviewer: ```suggestion
Setting up Threema Gateway is a two-step process:
```

Author: Updated docs according to new PR

> 1. **Generate encryption keys**: The integration generates a public and private key pair. You then need to provide the public key when creating your Gateway ID on the Threema Gateway website. As of March 2026, this requires manual approval by Threema, which may take some time.

Reviewer: ```suggestion
1. Generate encryption keys: 
   - The integration generates a public and private key pair. 
   - You need to provide the public key when creating your Gateway ID on the Threema Gateway website. 
      - As of March 2026, this requires manual approval by Threema, which may take some time.
```
Bold is used to markup UI strings, not to highlight text. I would use a nested list here.

> 2. **Configure the integration**: Once your Gateway ID is approved, use the Gateway ID, API secret, and the previously generated keys to complete the integration setup in Home Assistant.

Reviewer: ```suggestion
2. Configure the integration: 
   - Once your Gateway ID is approved, use the Gateway ID, API secret, and the previously generated keys to complete the integration setup in Home Assistant.
```

## Setup

> 

Reviewer: ```suggestion
```

During setup, you can choose between two options:

- **Create a new Gateway ID**: The integration generates an encryption key pair for you. Save the displayed keys — they cannot be recovered later. Then enter the Gateway ID and API secret from the Threema Gateway dashboard.

> - **Use an existing Gateway ID**: Enter your Gateway ID, API secret, and optionally your private and public keys for end-to-end encryption.

Reviewer: ```suggestion
- **Generate new encryption keys**: Creates a new key pair for a new Gateway ID.
- **Add existing Gateway ID**: Use an existing Gateway ID and API secret.
```

{% include integrations/config_flow.md %}

> 

Reviewer: ```suggestion
{% configuration_basic %}
Gateway ID:
  description: "Your Threema Gateway ID (starts with `*` and is 8 characters total)."
API secret:
  description: "The API secret from your Threema Gateway dashboard."
Private key:
  description: "Optional for basic mode, required for end-to-end encryption mode."
Public key:
  description: "Optional for basic mode, required for end-to-end encryption mode."
{% endconfiguration_basic %}
```
You can use a configuration_basic block here, see https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L57

Please double-check the content in my suggestion. I was just trying to give an example of what the block could look like.

Author: seems legit, even though I strongly recommend e2e, as this is why i wrote the integration. Other could also be a simple rest call.

Reviewer: Makes sense. Maybe better to rephrase to:
 
 description: "The public key required for end-to-end encryption mode.", 

to remove the focus on "optional"?

## QR code entity

When a public key is configured, the integration creates an **image entity** displaying a QR code. This QR code can be scanned by Threema users to verify the gateway identity. The QR code encodes the Gateway ID and public key in the format used by the Threema app.

## Actions

> ### Action `threema.send_message`

Reviewer: is this still accurate? currently, I don't see send_message in core.

Author: Updated docs according to new PR

Send a text message to a Threema user.

| Data attribute | Optional | Description |
| -------------- | -------- | ----------- |
| `config_entry_id` | yes | The config entry to use. Auto-selected if only one is configured. |
| `recipient` | no | The 8-character Threema ID of the recipient. |
| `message` | no | The text message to send (max 3,500 characters). |

### Examples

#### Send a simple text message

```yaml
action: threema.send_message
data:
  recipient: "YOUR_THREEMA_ID"
  message: "The front door was just opened!"
```

#### Send a message from an automation

```yaml
alias: "Notify on door open"
triggers:
  - trigger: state
    entity_id: binary_sensor.front_door
    to: "on"
actions:
  - action: threema.send_message
    data:
      recipient: "YOUR_THREEMA_ID"
      message: "Front door opened!"
```

#### Send a message with a template

{% raw %}

```yaml
action: threema.send_message
data:
  recipient: "YOUR_THREEMA_ID"
  message: "Temperature is {{ states('sensor.temperature') }}°C"
```

{% endraw %}

## Known limitations

- **Text messages only** — images, files, and other media are not supported.
- **No group messaging** — only 1-to-1 messages are supported.

> - **Send only** — receiving messages is not supported.

Reviewer: ```suggestion
- **Send only** — receiving messages is not supported.

## Removing the integration

This integration follows standard integration removal.

{% include integrations/remove_device_service.md %}
```
Does the integration removal follow the standard steps? Are there additional steps I should consider after removing the integration?

Author: Delete the Gateway. But I don't see this as necessary for HA docs.
